src/dscanner/analysis/run.d

```diff
@@ -229,10 +231,32 @@ void generateSonarQubeGenericIssueDataReport(string[] fileNames, const StaticAna
 bool analyze(string[] fileNames, const StaticAnalysisConfig config, string errorFormat,
 		ref StringCache cache, ref ModuleCache moduleCache, bool staticAnalyze = true)
 {
+	import dmd.parse : Parser;
+	import dmd.astbase : ASTBase;
+	import dmd.id : Id;
+	import dmd.globals : global;
+	import dmd.identifier : Identifier;
+	import std.string : toStringz;
+
+	Id.initialize();
+	global._init();
+	global.params.useUnitTests = true;
+	ASTBase.Type._init();
+
+
 	bool hasErrors;
 	foreach (fileName; fileNames)
 	{
 		auto code = readFile(fileName);
+
+		auto id = Identifier.idPool(fileName);
+		auto ast_m = new ASTBase.Module(fileName.toStringz, id, false, false);
+		auto input = cast(char[]) code;
+		input ~= '\0';
```

Comment:
I don't think this is needed as `readFile` uses `std.File.read()`.
I think it would be best to use [std.conv.to](https://dlang.org/phobos/std_conv.html#.to) in order to convert `code` (of type `ubyte[]`) to `input` (of type `char[]`)

Reply:
The problem is not the conversion from `ubyte[]` to `char[]`, but the fact that the lexer expects a `char*`.